src/main/kotlin/org/opensearch/replication/task/shard/TranslogSequencer.kt

```diff
@@ -62,13 +64,19 @@ class TranslogSequencer(scope: CoroutineScope, private val replicationMetadata:
                                                          leaderAlias, leaderIndexName)
                 replayRequest.parentTask = parentTaskId
                 launch {
+                    var relativeStartNanos  = System.nanoTime()
                     val replayResponse = client.suspendExecute(replicationMetadata, ReplayChangesAction.INSTANCE, replayRequest)
                     if (replayResponse.shardInfo.failed > 0) {
                         replayResponse.shardInfo.failures.forEachIndexed { i, failure ->
                             log.error("Failed replaying changes. Failure:$i:$failure")
                         }
-                        throw org.opensearch.replication.ReplicationException("failed to replay changes", replayResponse.shardInfo.failures)
+                        followerClusterStats.stats[followerShardId]!!.opsWriteFailures.addAndGet(replayResponse.shardInfo.failed.toLong())
+                        throw ReplicationException("failed to replay changes", replayResponse.shardInfo.failures)
                     }
+
```

Comment:
Do we need to handle when replay throws exception?

Reply:
Yes, this is not useful now. But once we have appropriate retry strategy , this would come in handy.